sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,436 @@
+# Form Recognizer Design Overview
+
+The Form Recognizer client library provides two clients to interact with the service: `FormRecognizerClient` and 
+`CustomFormClient`, which can be imported from the `azure.ai.formrecognizer` namespace. The asynchronous clients 
+can be imported from `azure.ai.formrecognizer.aio`.
+
+`FormRecognizerClient` provides methods for interacting with the prebuilt models (receipt and layout).
+`CustomFormClient` provides the methods for training custom models to analyze forms.
+
+Authentication is achieved by passing an instance of `CognitiveKeyCredential("<api_key>")` to the client,
+or by providing a token credential from `azure.identity` to use Azure Active Directory.
+
+## Prebuilt
+
+The prebuilt models are accessed through the `FormRecognizerClient`. The input form or document can be passed as a 
+string url/path to the image, or as a file stream. The SDK will determine content-type and send the appropriate header. 
+
+Both prebuilt methods return poller objects which are used to get the result.
+The `begin_extract_receipt` method returns an `ExtractedReceipt` with hardcoded receipt fields.
+The `begin_extract_layout` method returns the extracted tables in a tabular format such that the user can
+index into a specific row or column and easily integrate with other Python libraries.
+
+If the keyword argument `include_text_details=True` is passed in, the `raw_` attributes will be populated with the
+raw OCR result for each value/cell.
+
+### Form Recognizer Client
+```python
+from azure.ai.formrecognizer import FormRecognizerClient
+
+client = FormRecognizerClient(endpoint: str, credential: Union[CognitiveKeyCredential, TokenCredential])
+
+client.begin_extract_receipt(form: Any, **kwargs) -> LROPoller -> List[ExtractedReceipt]
+
+client.begin_extract_layout(form: Any, **kwargs) -> LROPoller -> List[ExtractedLayoutPage]
+```
+
+### Receipt Models
+```python
+class ExtractedReceipt:
+    receipt_items: List[ReceiptItem]
+    merchant_address: str
+    merchant_name: str
+    merchant_phone_number: str
+    receipt_type: str
+    subtotal: float
+    tax: float
+    tip: float
+    total: float
+    transaction_date: str
+    transaction_time: str
+    fields: ReceiptFields
+
+class ReceiptItem:
+    name: str
+    quantity: int
+    total_price: float
+
+class ReceiptFields:
+    receipt_items: List[ReceiptItemField]
+    merchant_address: FieldValue
+    merchant_name: FieldValue
+    merchant_phone_number: FieldValue
+    receipt_type: str
+    subtotal: FieldValue
+    tax: FieldValue
+    tip: FieldValue
+    total: FieldValue
+    transaction_date: FieldValue
+    transaction_time: FieldValue
+
+class ReceiptItemField:
+    name: str
+    quantity: int
+    total_price: float
+    bounding_box: List[float]
+    confidence: float
+    raw_field: List[ExtractedLine]
+
+class FieldValue:
+    value: Union[str, float, int]
+    bounding_box: List[float]
+    confidence: float
+    raw_field: List[ExtractedLine]
```

Comment:
Will re-think naming on this one.

I'm hoping to return a homogeneous list here to simplify things. My understanding of the readResult response is that  the `elements` field references a Line which is composed of Words. When would I only need to return the Words and not the Line they came from?

Reply:
The `elements` field may reference a Line or a Word (word is more common currently).

Think of a hypothetical receipt line item. The line text might be:
`1  $5.00  Slice of Pizza`

For fields Quantity, Price, and Name, respectively.

The words in that line would be:
`["1", "$5.00", "Slice", "of", "Pizza"]`

So the `Name` field would have an `elements` array of:
```json
[
  "#/readResults/0/lines/0/words/2",
  "#/readResults/0/lines/0/words/3",
  "#/readResults/0/lines/0/words/4",
]
```

If the Name referenced the entire line, it would point to extra/incorrect information related to other fields.

Reply:
Very clear, thanks for the example. Will update this accordingly